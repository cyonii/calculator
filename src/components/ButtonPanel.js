// import { ReactFragment } from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = (props) => {
  const handleBtnClick = (buttonName) => props.btnClickHandler(buttonName);

  const renderBtn = (name, colClass, btnClass = '') => (
    <div className={`${colClass}`}>
      <Button className={btnClass} name={name} clickHandler={handleBtnClick} />
    </div>
  );

  return (
    <div data-testid="button-panel">
      <div className="row mb-3">
        {renderBtn('AC', 'col-3')}
        {renderBtn('+/-', 'col-3')}
        {renderBtn('%', 'col-3')}
        {renderBtn('รท', 'col-3', 'btn-primary border-primary')}
      </div>

      <div className="row mb-3">
        {renderBtn('7', 'col-3')}
        {renderBtn('8', 'col-3')}
        {renderBtn('9', 'col-3')}
        {renderBtn('*', 'col-3', 'btn-primary border-primary')}
      </div>

      <div className="row mb-3">
        {renderBtn('4', 'col-3')}
        {renderBtn('5', 'col-3')}
        {renderBtn('6', 'col-3')}
        {renderBtn('-', 'col-3', 'btn-primary border-primary')}
      </div>

      <div className="row mb-3">
        {renderBtn('1', 'col-3')}
        {renderBtn('2', 'col-3')}
        {renderBtn('3', 'col-3')}
        {renderBtn('+', 'col-3', 'btn-primary border-primary')}
      </div>

      <div className="row">
        {renderBtn('0', 'col-6')}
        {renderBtn('.', 'col-3')}
        {renderBtn('=', 'col-3', 'btn-primary border-primary')}
      </div>
    </div>
  );
};

ButtonPanel.propTypes = {
  btnClickHandler: PropTypes.func.isRequired,
};

export default ButtonPanel;
