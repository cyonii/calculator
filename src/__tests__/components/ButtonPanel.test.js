import { render, screen, cleanup } from '@testing-library/react';
import ButtonPanel from '../../components/ButtonPanel';

describe('<ButtonPanel /> Component', () => {
  let testBtnPanel;

  beforeEach(async () => {
    testBtnPanel = <ButtonPanel btnClickHandler={jest.fn} />;
  });

  afterEach(() => cleanup);

  test('should render correctly', () => {
    render(<ButtonPanel btnClickHandler={jest.fn} />);

    const buttonPanel = screen.getByTestId('button-panel');
    expect(buttonPanel).toBeInTheDocument();
    expect(buttonPanel.querySelectorAll('button')).toHaveLength(19);
  });

  test('should match snapshot', () => {
    expect(testBtnPanel).toMatchSnapshot();
  });
});
