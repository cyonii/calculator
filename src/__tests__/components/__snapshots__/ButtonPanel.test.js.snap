// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonPanel /> Component should match snapshot 1`] = `
<ButtonPanel
  btnClickHandler={[Function]}
/>
`;
